pepper-apis/src/main/java/org/broadinstitute/ddp/service/participantslookup/ParticipantsLookupService.java

```diff
@@ -0,0 +1,79 @@
+package org.broadinstitute.ddp.service.participantslookup;
+
+
+import static org.broadinstitute.ddp.service.participantslookup.error.ParticipantsLookupErrorType.ELASTIC_SEARCH_STATUS;
+import static org.broadinstitute.ddp.service.participantslookup.error.ParticipantsLookupErrorType.INVALID_QUERY;
+import static org.broadinstitute.ddp.service.participantslookup.error.ParticipantsLookupErrorType.INVALID_RESULT_MAX_COUNT;
+import static org.broadinstitute.ddp.service.participantslookup.error.ParticipantsLookupErrorType.STUDY_GUID_UNKNOWN;
+
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.service.participantslookup.error.ParticipantsLookupException;
+import org.elasticsearch.ElasticsearchStatusException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract class defining a service for Pepper participants lookup.<br>
+ * It implements a backend functionality for PRISM participants lookup component.<br>
+ * It can have different implementations (i.e. participants searching could
+ * be done in different sources: DB, ElasticSearch, etc.).<br>
+ * Currently supported only searching in Pepper ElasticSearch database.
+ */
+public abstract class ParticipantsLookupService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ParticipantsLookupService.class);
+
+    private static final int QUERY_MAX_LENGTH = 100;
+
+    /**
+     * Lookup participants (it can return more than 1 results).
+     * The results count is limited by parameter 'resultsMaxCount': so, this method could return no more
+     * than 'resultMaxCount' rows. The real count of rows which are found is saved into result object
+     * in {@link ParticipantsLookupResult#getTotalCount()}.
+     *
+     * @param studyGuid GUID of a study where to search participants
+     * @param query string containing a fragment by which to lookup the participants (do a full-text search).
+     * @return ParticipantsLookupResult - an object containing the participants lookup results.
+     */
+    public ParticipantsLookupResult lookupParticipants(String studyGuid, String query, int resultsMaxCount) {
+
+        if (studyGuid == null) {
+            throw new ParticipantsLookupException(STUDY_GUID_UNKNOWN, "studyGuid cannot be null");
```

Comment:
How about we query for the study in the route and:
* If good, then pass in the `StudyDto`, so then you don't have to query for it in `detectEsIndices()`;
* If missing, halt with `400 Not Found`.

Reply:
Done